src/Options/OptionsDetails.hpp

```diff
@@ -184,14 +185,15 @@ struct print_impl<Tag, OptionList,
                  << pretty_type::short_name<decltype(*derived)>();
             });
       } else {
-        ss << "\n    default=" << std::boolalpha << Tag::default_value();
+        ss << "\n    default="
+           << (MakeString{} << std::boolalpha << Tag::default_value());
       }
     }
     if constexpr (has_lower_bound<Tag>::value) {
-      ss << "\n    min=" << Tag::lower_bound();
+      ss << "\n    min=" << (MakeString{} << Tag::lower_bound());
     }
     if constexpr (has_upper_bound<Tag>::value) {
-      ss << "\n    max=" << Tag::upper_bound();
+      ss << "\n    max=" << (MakeString{} << Tag::upper_bound());
```

Comment:
For my understanding, why is this change needed?

Reply:
I would guess it tricks ADL into looking into the global namespace. I'm unsure if a `using ::operator<<;` would work here because that might depend on include order -_-

Reply:
> For my understanding, why is this change needed?

I don't know.  I don't understand how overload resolution interacts with namespaces (except for the basic rule that the namespaces containing the argument types are searched).  The original code stopped working, and this worked.

`using ::operator<<;` fails to compile on some files because it finds nothing to import.

Reply:
I think the rule is that since `MakeString` is in the global namespace and `std::BLAH` is in the `std` namespace `operator<<` is searched for in both the global and `std` namespace. Since we can't add `operator<<` for `std` things into the `std`namespace, using `MakeString` gets the compiler to search in the global namespace, which is where we have `operator<<` defined for the STL containers. I think the reason `using operator<<` fails is because it's eager while the `MakeString` trick causes the compiler to search for the stream operator when the function is used, not when it is defined :roll_eyes: